internal/dag/httpproxy_processor.go

```diff
@@ -516,6 +516,48 @@ func (p *HTTPProxyProcessor) computeHTTPProxy(proxy *contour_api_v1.HTTPProxy) {
 	}
 	insecure.CORSPolicy = cp
 
+	var (
```

Comment:
Please add some new unit test cases for vhost-level header policies to `builder_test.go` in this package. You should be able to find existing test cases for route-level header policies and more or less repeat them. Let me know if you need further direction.

Reply:
Test cases are added